demos/storage-import-zipped-vhds/prereqs/prereq.azuredeploy.json

```diff
@@ -233,7 +199,7 @@
         },
         "uriWritableStorageAccountBlobContainerSasToken": {
             "type": "string",
-            "value": "[reference(variables('deploymentScriptNameCreateDestination')).outputs.uriWritableStorageAccountBlobContainerSasToken]"
+            "value" : "[concat(reference(parameters('storageAccountNameDestination')).primaryEndpoints.blob, parameters('blobContainerNameDestination'), '?', listAccountSas(parameters('storageAccountNameDestination'), '2021-04-01', variables('listAccountSasRequestContent')).accountSasToken)]"
```

Comment:
you cannot pass a secret through an output - two options you have
1) put the secret into keyvault and pass a reference (you would have to create the keyvault in prereqs)
2) output the storageAccount resourceId (and blob info) and call listAccountSas() in azuredeploy.json

Reply:
Thanks, option 1 is really nice. Both Source and Destination prereqs are now passing securely via KeyVault. I tested preqs and main template in my own sub and it worked well.

Now there is a little glitch. The parameters file does not like the GET-PREREQ in the attribute reference.keyvault.id used like this:

        "Source": {
            "reference": {
                "keyVault": {
                    "id" : "GET-PREREQ-KeyVaultId"
                },
                "secretName": "Source"
            }
        },

In the pipeline the Best Practices Tests raises following errors:

'Source'.reference.keyVault.id contains improper value, 
 Should be like : /subscriptions/<subscription id>/resourceGroups/<resource group name>/providers/Microsoft.KeyVault/vaults/<key vault name>
'Destination'.reference.keyVault.id contains improper value, 
 Should be like : /subscriptions/<subscription id>/resourceGroups/<resource group name>/providers/Microsoft.KeyVault/vaults/<key vault name>

And the Deploy mainTemplate fails with 

Code=KeyVaultParameterReferenceInvalidResourceId; Message=The resource identifier of the KeyVault parameter 'Destination' is invalid. Please specify the value following '/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.KeyVault/vaults/{vaultName}'